lib/octokit/client/authorizations.rb

```diff
@@ -1,185 +0,0 @@
-          end
-
-          # Remove the client_id from the body otherwise
-          # this will result in a 422.
-          options.delete(:client_id)
-
-          if (fingerprint = options.delete(:fingerprint))
-            put "authorizations/clients/#{client_id}/#{fingerprint}", options.merge(client_secret: client_secret)
-          else
-            put "authorizations/clients/#{client_id}", options.merge(client_secret: client_secret)
-          end
-
-        else
-          post 'authorizations', options
-        end
-      end
-
-      # Update an authorization for the authenticated user.
-      #
-      # You can update your own tokens, but only through
-      # Basic Authentication.
-      #
-      # @param options [Hash] A customizable set of options.
-      # @option options [Array] :scopes Replace the authorization scopes with these.
-      # @option options [Array] :add_scopes A list of scopes to add to this authorization.
-      # @option options [Array] :remove_scopes A list of scopes to remove from this authorization.
-      # @option options [String] :note A note to remind you what the OAuth token is for.
-      # @option options [String] :note_url A URL to remind you what app the OAuth token is for.
-      #
-      # @return [Sawyer::Resource] A single (updated) authorization for the authenticated user
-      # @see https://developer.github.com/v3/oauth_authorizations/#update-an-existing-authorization
-      # @see https://developer.github.com/v3/oauth/#scopes for available scopes
-      # @example Update the authorization for user ctshryock's project Zoidberg
-      #  client = Octokit::Client.new(:login => 'ctshryock', :password => 'secret')
-      #  client.update_authorization(999999, {:add_scopes => ["gist", "repo"], :note => "Why not Zoidberg possibly?"})
-      def update_authorization(number, options = {})
-        patch "authorizations/#{number}", options
-      end
-
-      # Delete an authorization for the authenticated user.
-      #
-      # You can delete your own tokens, and only through
-      # Basic Authentication.
-      #
-      # @param number [Number] An existing Authorization ID
-      #
-      # @return [Boolean] Success
-      # @see https://developer.github.com/v3/oauth_authorizations/#delete-an-authorization
-      # @example Delete an authorization
-      #  client = Octokit::Client.new(:login => 'ctshryock', :password => 'secret')
-      #  client.delete_authorization(999999)
-      def delete_authorization(number, options = {})
-        boolean_from_response :delete, "authorizations/#{number}", options
-      end
-
-      # Check scopes for a token
-      #
-      # @param token [String] GitHub OAuth token
-      # @param options [Hash] Header params for request
-      # @return [Array<String>] OAuth scopes
-      # @see https://developer.github.com/v3/oauth/#scopes
-      def scopes(token = @access_token, options = {})
```

Comment:
It might be worth keeping this one? 🤔 It isn't specifically related to the Authorizations API and it could be useful for users. I'm not sure we have any replacement to it either, as I don't think there is a way to access headers in the SDK.

Reply:
Good point! Would you rather I keep the authorizations module around for this use case, or refactor this functionality somewhere else?

Reply:
Kinda up to you, honestly. I would probably move it ("authorizations" isn't really a sensible name, since I think it works for any token?).